### DISCUSSION TITLE
PHP Error, core Error and fulltextsearch warning in version 8.3.0 Beta

### DISCUSSION THREAD

**Discussion by Frequency-Overload** [nextcloud/all-in-one#4756] [Questions] [answered]
### Overview
Testing Nextcloud Hub 8 (29.0.1) with the latest beta version. Several Errors and Warnings were logged in the “Logging” section on the Nextcloud Interface. 

From within the Nextcloud interface and doing a search for a file to creating a new document; these warnings or errors haven’t affected this.

As a side note I tried out ClarmAV by uploading EICAR.txt, and the file was detected.

### Administration Settings -> Logging -> Log reader

>**Error: PHP**
>Optional parameter $userId declared before required parameter $path is implicitly          treated as a required parameter at /var/www/html/custom_apps/deck/lib/Service/ConfigService.php#234

>**Error: PHP**
>fopen(/var/www/html/config/config.php): Failed to open stream: No such file or directory at /var/www/html/lib/private/Config.php#225

>**Error: core**
> **NotFoundException** /appdata_ocec5n33fl9g/theming/global
Error while running background job OCA\Theming\Jobs\MigrateBackgroundImages (id: 60, arguments: {"stage":"prepare"}) 

>**Warning: fulltextsearch_elasticsearch**
> **ClientResponseException** 404 Not Found: {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [nextcloud-aio]","resource.type":"index_or_alias","resource.id":"nextcloud-aio","index_uuid":"_na_","index":"nextcloud-aio"}],"type":"index_not_found_exception","reason":"no such index [nextcloud-aio]","resource.type":"index_or_alias","resource.id":"nextcloud-aio","index_uuid":"_na_","index":"nextcloud-aio"},"status":404}

>**Warning: fulltextsearch_elasticsearch**
>**ClientResponseException** 404 Not Found: {"error":{"root_cause":[{"type":"resource_not_found_exception","reason":"pipeline [attachment] is missing"}],"type":"resource_not_found_exception","reason":"pipeline [attachment] is missing"},"status":404}

### Debian_Version: 12.5  
kernel version : 6.1.0-21-amd64
Codename: bookworm

#### Other valuable info 
The container log for nextcloud-aio-nextcloud mentions about PHP being deprecated
[nextcloud-aio-nextcloud log.txt](https://github.com/user-attachments/files/15529440/nextcloud-aio-nextcloud.log.txt)

#### **PHP ERROR - 234**
![PHP # 234](https://github.com/nextcloud/all-in-one/assets/168395765/93b98042-fac5-4e67-b0ac-9d7d8ce54c43)

#### **PHP ERROR -  225**
![PHP #  225](https://github.com/nextcloud/all-in-one/assets/168395765/5db59b8c-df0a-4f5c-8f37-eb2da271e7c0)

#### **CORE ERROR**
![core - NotFoundException](https://github.com/nextcloud/all-in-one/assets/168395765/636d8a7c-b316-4a0b-9592-f1bf429b39ed)

#### **Warning fulltextsearch_elasticsearch**
![warning - fulltextsearch_elasticsearch](https://github.com/nextcloud/all-in-one/assets/168395765/15ed4546-e4b0-430a-9f02-a4424703313f)

#### **ClamAV Test**
![ClamAV Test](https://github.com/nextcloud/all-in-one/assets/168395765/883f5447-7c8c-4f94-ba5b-84b076a1fb78)

**Comment by szaimen** [comment, marked as answer]
Hi,

> > **Error: PHP**
> > Optional parameter $userId declared before required parameter $path is implicitly          treated as a required parameter at /var/www/html/custom_apps/deck/lib/Service/ConfigService.php#234

This needs to be fixed in : https://github.com/nextcloud/deck/issues

> > **Error: PHP**
> > fopen(/var/www/html/config/config.php): Failed to open stream: No such file or directory at /var/www/html/lib/private/Config.php#225

This needs to be reported to and fixed in https://github.com/nextcloud/server/issues

> > **Error: core**
> > **NotFoundException** /appdata_ocec5n33fl9g/theming/global
> > Error while running background job OCA\Theming\Jobs\MigrateBackgroundImages (id: 60, arguments: {"stage":"prepare"})

This needs to be reported to and fixed in https://github.com/nextcloud/server/issues

> > **Warning: fulltextsearch_elasticsearch**
> > **ClientResponseException** 404 Not Found: {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [nextcloud-aio]","resource.type":"index_or_alias","resource.id":"nextcloud-aio","index_uuid":"_na_","index":"nextcloud-aio"}],"type":"index_not_found_exception","reason":"no such index [nextcloud-aio]","resource.type":"index_or_alias","resource.id":"nextcloud-aio","index_uuid":"_na_","index":"nextcloud-aio"},"status":404}
> 
> > **Warning: fulltextsearch_elasticsearch**
> > **ClientResponseException** 404 Not Found: {"error":{"root_cause":[{"type":"resource_not_found_exception","reason":"pipeline [attachment] is missing"}],"type":"resource_not_found_exception","reason":"pipeline [attachment] is missing"},"status":404}

Hm, I cannot reproduce this. Can you follow https://github.com/nextcloud/all-in-one/discussions/2509 and check if that helps?

**Comment by Frequency-Overload** [comment]
I did a full reset of the nextcloud instance. The fulltextsearch_elasticsearch warning only ever appears after the first time of the install. The error does not appear again after using the program. 

I then tried doing another full reset, but this time before opening and logging in for the first time I followed https://github.com/nextcloud/all-in-one/discussions/2509.  Then logged in and checked the “Log reader” and the fulltextsearch_elasticsearch warnings were still there. 

This warning only appears very earlier on, and does not show up again.

I should also mention that when I install debian, I encrypt the root volume.
